webdriver/tests/bidi/browser/create_user_context/unhandled_prompt_behavior.py

```diff
@@ -0,0 +1,96 @@
+import asyncio
+import pytest
+
+from webdriver.bidi.modules.script import ContextTarget
+from ... import recursive_compare
+
+pytestmark = pytest.mark.asyncio
+
+USER_PROMPT_OPENED_EVENT = "browsingContext.userPromptOpened"
+
+
+@pytest.mark.parametrize("handler", ["accept", "dismiss", "ignore"])
+async def test_beforeunload(bidi_session, subscribe_events, wait_for_event,
+        wait_for_future_safe, create_user_context, handler,
+        setup_beforeunload_page, inline):
+    user_context = await create_user_context(
+        unhandled_prompt_behavior={"beforeUnload": handler})
+    new_tab = await bidi_session.browsing_context.create(
+        type_hint="tab",
+        user_context=user_context
+    )
+
+    await setup_beforeunload_page(new_tab)
+
+    await subscribe_events(events=[USER_PROMPT_OPENED_EVENT])
+    on_entry = wait_for_event(USER_PROMPT_OPENED_EVENT)
+
+    url_after = inline("<div>foo</div>")
+
+    asyncio.create_task(
+        bidi_session.browsing_context.navigate(context=new_tab["context"],
+                                               url=url_after, wait="none"))
+
+    # Wait for prompt to appear.
+    resp = await wait_for_future_safe(on_entry)
+    recursive_compare({
+        'context': new_tab["context"],
+        'handler': handler,
+        'type': "beforeunload",
+    }, resp)
+
+
+@pytest.mark.parametrize("handler", ["accept", "dismiss", "ignore"])
+@pytest.mark.parametrize("prompt", ["alert", "confirm", "prompt", "default"])
+async def test_simple_prompts(bidi_session, subscribe_events, wait_for_event,
+        wait_for_future_safe, create_user_context, prompt, handler):
+    prompt_type = prompt if (prompt is not None and prompt != 'default') else 'alert'
```

Comment:
When reading this line I always wrap my head around why we are falling back to `alert`. Maybe you can add a comment and explain it? It would help in the future.